Slim/Networking/Async/HTTP.pm

```diff
@@ -100,49 +116,51 @@ sub new_socket {
 			# DNS lookup.
 			# So we will probably need to explicitly set "SSL_hostname" if we are to succeed with such
 			# a server.
-
+			
 			# First, try without explicit SNI, so we don't inadvertently break anything. 
 			# (This is the 'old' behaviour.) (Probably overly conservative.)
-			my $sock = Slim::Networking::Async::Socket::HTTPS->new( @_ );
+			
+			if (%socks) {
+				$sock = Slim::Networking::Async::Socket::HTTPSSocks->new( %args, %socks );
+				$sock->blocking(0);
```

Comment:
Couldn't this line be in `->new()`? This would avoid having to remember to call it after each instantiation.